src/mnode/rpc/masternode.cpp

```diff
@@ -117,6 +127,10 @@ UniValue masternodelist(const UniValue& params, bool fHelp)
         std::map<COutPoint, CMasternode> mapMasternodes = masterNodeCtrl.masternodeManager.GetFullMasternodeMap();
         for (auto& mnpair : mapMasternodes) {
             CMasternode mn = mnpair.second;
+            if( mn.IsNewStartRequired() && ! mn.IsPingedWithin(masterNodeCtrl.MNStartRequiredExpirationTime) && strExtra != "allnode") 
```

Comment:
you can calculate visibility filter once outside the loop and use it here:
const bool bShowAllNodes = strExtra == "allnode";